dockers/docker-orchagent/ipinip_ping.py

```diff
@@ -0,0 +1,197 @@
+#! /usr/bin/env python3
+"""
+Adds neighbor to kernel for undeliverable IPinIP packets
+
+When receiving IPinIP packets, if the hardware doesn't contain neighbor
+information for the inner packet's destination IP, the entire encapsulated
+packet is trapped to the CPU. In this case, we should ping the inner
+destination IP to trigger the process of obtaining neighbor information
+"""
+import subprocess
+import time
+
+from datetime import datetime
+from scapy.layers.inet import IP
+from scapy.layers.inet6 import IPv6
+from scapy.sendrecv import sniff
+from swsssdk import ConfigDBConnector
+from sonic_py_common import logger as log
+
+logger = log.Logger()
+
+
+class IPinIPListener(object):
+
+    def __init__(self):
+        self.config_db = ConfigDBConnector()
+        self.config_db.connect()
+
+    def get_portchannel_members(self):
+        """
+        Gets all interfaces which are portchannel members in config DB
+
+        Returns:
+            (list) Interface names which are portchannel members, stored
+                   as strings
+        """
+        raw_members = self.config_db.get_keys('PORTCHANNEL_MEMBER')
+        members = [pair[1] for pair in raw_members]
+        return members
+
+    def all_ports_ok(self):
+        """
+        Checks if the host interfaces corresponding to portchannel
+        members are up
+
+        Returns:
+            (bool) True, if all interfaces are up
+                   False, otherwise
+        """
+        members = self.get_portchannel_members()
+        for intf in members:
+            cmds = ['ip', 'link', 'show', intf]
+            try:
+                output = subprocess.check_output(cmds)
+            except subprocess.CalledProcessError:
+                # This error will be raised if the host interface has
+                # not been created yet and the `ip link` command fails
+                return False
+            if 'state UP' not in str(output):
+                return False
+        return True
+
+    def wait_for_ports_up(self, interval=5, timeout=60):
```

Comment:
This will be tricky as interfaces may stay `down `or even sometimes `unknown` in kernel. IMO, lets use the STATE_DB to check the port state. You could refer [here](https://github.com/Azure/sonic-buildimage/blob/master/dockers/docker-dhcp-relay/wait_for_intf.sh.j2#L38)

Reply:
Sure, will change. But if the interface is `down` or `unknown`, will scapy even be able to listen on that interface?